### PR TITLE
Add example implementation of card component

### PR DESCRIPTION
**Pull request by OscarSpruit**

### CODE REVIEW

File: example-app/src/main/java/com/adyen/checkout/example/ui/card/CardViewState.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.adyen.checkout.example.ui.card
+
+import com.adyen.checkout.components.model.paymentmethods.PaymentMethod
+
+internal sealed class CardViewState {
+
+    object Loading : CardViewState()
+
+    data class Data(val paymentMethod: PaymentMethod) : CardViewState()
```
Comment (caiofaustino) [inline example-app/src/main/java/com/adyen/checkout/example/ui/card/CardViewState.kt:9]: Maybe something more expressive like `ShowComponent` would be better?

File: example-app/src/main/java/com/adyen/checkout/example/ui/card/CardFragment.kt [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    private fun setupCardView(paymentMethod: PaymentMethod) {
+        val cardComponent = CardComponent.PROVIDER.get(
+            this,
+            paymentMethod,
+            checkoutConfigurationProvider.getCardConfiguration()
+        )
+
+        binding.cardView.attach(cardComponent, viewLifecycleOwner)
+
+        cardComponent.observe(viewLifecycleOwner, cardViewModel::onCardComponentState)
```
Comment (caiofaustino) [inline example-app/src/main/java/com/adyen/checkout/example/ui/card/CardFragment.kt:83]: Maybe we should also add `observeError` to illustrate a possible failure scenario.

**Review by caiofaustino — DISMISSED** [review]

File: example-app/src/main/java/com/adyen/checkout/example/ui/card/CardFragment.kt (hunk 2 above)
Comment (OscarSpruit) [reply example-app/src/main/java/com/adyen/checkout/example/ui/card/CardFragment.kt:83]: Good one! Wasn't aware of this separate error stream. Didn't see it in the docs as well, should it be mentioned there? https://docs.adyen.com/online-payments/android/components

File: example-app/src/main/java/com/adyen/checkout/example/ui/card/CardFragment.kt [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        binding.payButton.setOnClickListener { cardViewModel.onPayClick() }
+
+        lifecycleScope.launchWhenStarted {
+            launch { cardViewModel.cardViewState.collect(::onCardViewState) }
+            launch { cardViewModel.paymentResult.collect(::onPaymentResult) }
+        }
+    }
+
+    private fun onCardViewState(cardViewState: CardViewState) {
+        when (cardViewState) {
```
[the remark this answers is not in the document]
Comment (OscarSpruit) [reply example-app/src/main/java/com/adyen/checkout/example/ui/card/CardFragment.kt:50]: Could be, but I prefer it this way. Gives you a bit more control over the different view states

File: example-app/src/main/java/com/adyen/checkout/example/ui/card/CardViewState.kt (hunk 1 above)
Comment (OscarSpruit) [reply example-app/src/main/java/com/adyen/checkout/example/ui/card/CardViewState.kt:9]: Good suggestion

**Review by caiofaustino — APPROVED** [review]

**Pull request merged**